packages/nextjs-mf/src/plugins/JsonpChunkLoading.ts

```diff
@@ -0,0 +1,314 @@
+// CustomWebpackPlugin.ts
+import type {
+  Chunk,
+  Compilation,
+  Compiler,
+  Module,
+  RuntimeModule,
+} from 'webpack';
+import { ConcatSource } from 'webpack-sources';
+// @ts-ignore
+import JsonpChunkLoadingRuntimeModule from 'webpack/lib/web/JsonpChunkLoadingRuntimeModule';
+import Template from '../../utils/Template';
+import { DEFAULT_SHARE_SCOPE_BROWSER } from '../internal';
+import { loadDependenciesTemplate } from './LoadDependenciesTemplate';
+
+// @ts-ignore
+export const setEagerLoading = (RuntimeGlobals) => {
+  return Template.asString([
+    'function setEagerLoading(libraryName, version, fromValue) {',
+    Template.indent([
+      'var library = ' +
+        RuntimeGlobals.shareScopeMap +
+        '.default[libraryName];',
+      'var shareScopeMap = ' + RuntimeGlobals.shareScopeMap + '.default;',
+      'var alternativeVersion = Object.keys(shareScopeMap[libraryName])[0];',
```

Comment:
Sometimes host has lower version of shared singleton, so the rules cannot be predictably determine. In this case (not eager), find the first key in the scope and load that.